sys/net/link_layer/Kconfig.lorawan

```diff
@@ -0,0 +1,518 @@
+# Copyright (c) 2020 Freie Universitaet Berlin
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+#
+config HAVE_LORAWAN
+    bool
+    help
+        Indicates that LoRaWAN support is present.
+
+menuconfig KCONFIG_USEMODULE_LORAWAN
+    bool "Configure LoRaWAN MAC"
+    depends on HAVE_LORAWAN
+    help
+        Configure LoRaWAN module using Kconfig. This may override configurations
+        in LoRa PHY.
+
+if KCONFIG_USEMODULE_LORAWAN
+
+choice
+    bool "Channel plan"
+    depends on USEMODULE_GNRC_LORAWAN || USEPKG_SEMTECH_LORAMAC
+    default LORAMAC_REGION_EU_868
+    help
+        Configure the LoRaWAN region.
+
+if USEPKG_SEMTECH_LORAMAC
+
+config LORAMAC_REGION_AS_923
+    bool "AS923"
+
+config LORAMAC_REGION_AU_915
+    bool "AU915-928"
+
+config LORAMAC_REGION_CN_470
+    bool "CN470-510"
+
+config LORAMAC_REGION_CN_779
+    bool "CN779-787"
+
+config LORAMAC_REGION_EU_433
+    bool "EU433"
+
+endif # USEPKG_SEMTECH_LORAMAC
+
+config LORAMAC_REGION_EU_868
+    bool "EU863-870"
+
+config LORAMAC_REGION_IN_865
+    bool "IN865-867"
+
+if USEPKG_SEMTECH_LORAMAC
+
+config LORAMAC_REGION_KR_920
+    bool "KR920-923"
+
+config LORAMAC_REGION_RU_864
+    bool "RU864-870"
+
+config LORAMAC_REGION_US_915
+    bool "US902-928"
+
+config LORAMAC_REGION_US_915_HYBRID
+    bool "US902-928(hybrid)"
+
+endif # USEPKG_SEMTECH_LORAMAC
+
+endchoice
+
+choice
+    bool "Join procedure"
+    depends on USEMODULE_GNRC_LORAWAN
+    default LORAMAC_DEFAULT_JOIN_PROCEDURE_OTAA
+    help
+        Configure the join procedure for end node activation. There are two
+        options, Over The Air Activation (OTAA) results in device sending
+        join request to the network whereas in the case of Activation By
+        Personalization (ABP) the user enters the activation keys manually. OTAA
+        is the preferred and most secure way to connect to a LoRaWAN network.
+        For quick testing ABP is preferred as the device can transmit right
+        away without waiting for the network to provision the keys.
+
+config LORAMAC_DEFAULT_JOIN_PROCEDURE_OTAA
+    bool "Over the air activation (OTAA)"
+
+config LORAMAC_DEFAULT_JOIN_PROCEDURE_ABP
+    bool "Activation by personalization (ABP)"
+
+endchoice
```

Comment:
During an offline discussion with @jia200x, we realized that this default join procedure option may be complicating things unnecessarily. Clearly it is an option specific for the GNRC LoRaWAN stack and is not needed on a 'standard' lorawan application. In the end it's up to the application to perform the join procedure.

I think we could make a menu containing all credentials (both OTAA and ABP) and let the user assign values to all of them, defaulting to `0000000...`. For user help comments or sub menus can be added to specify which are mandatory for each join type.

Reply:
Sure. AFAIK this configuration works for GNRC LoRaWAN stack the semantics is a bit confusing from an application point of view as you had suggested. As discussed a few weeks ago on a separate thread, should I move it to a new menu named "Application" in Kconfig? I can move these Macro definitions to application space. Does that make sense?

Reply:
No, the parameters are still a configuration of GNRC LoRaWAN, so maybe just keep them on it's Kconfig file. Unless @jia200x  has a different idea on this?

Reply:
I also think this should be kept in the stack configuration

Reply:
Okay. Maybe I will modify the header (loramac.h) to default to OTAA `ifndef LORAMAC_DEFAULT_JOIN_PROCEDURE_ABP && LORAMAC_DEFAULT_JOIN_PROCEDURE_OTAA` , just like in the case of region/channel plan. Does that sound fine ?

Reply:
> Okay. Maybe I will modify the header (loramac.h) to default to OTAA ifndef LORAMAC_DEFAULT_JOIN_PROCEDURE_ABP && LORAMAC_DEFAULT_JOIN_PROCEDURE_OTAA , just like in the case of region/channel plan. Does that sound fine ?

I didn't understand this. Could you elaborate?

Reply:
Fixup did as discussed.

Reply:
ping @akshaim

Reply:
@jia200x This [fixup](https://github.com/RIOT-OS/RIOT/pull/15507/commits/75c158615674057c37f8a1f681c906f55406e894) allows for configuring both ABP and OTAA credentials regardless of join procedure.

Reply:
@jia200x @leandrolanzieri  Okay to squash?

Reply:
Yes please, go ahead